### ISSUE TITLE
Ask user if 3D flood-fill result is sane before applying mask to canvas

### ISSUE THREAD

**Issue by hanslovsky** [saalfeldlab/paintera#302]
Users currently can revert catastrophic 3D flood fills by dropping the current canvas. This is an OK solution but it is sub-optimal because users will lose all current painting process. Instead, the 3D flood-fill should ask the users to confirm that the flood fill result should be applied to the canvas. Then, the mask could simply be discarded without any changes to the canvas if necessary! The dialog should be non-blocking to allow to browse the data and to refresh the meshes for an informed decision. Painting must be blocked until confirmation or cancellation. This is not necessary for the 2D flood-fill because users can see everything that will be filled in the 2D views.

**Comment by hanslovsky** [comment]
cc @cpatrickc

**Comment by igorpisarev** [comment]
3D flood-filling currently allows cancellation via the progress indicator menu or pressing ESC. That helps in situations such as if it accidentally starts filling the background.

**Comment by hanslovsky** [comment]
Yes, it does, but it should also be possible to cancel after the flood-fill has finished, e.g. when leaving the workstation because the flood-fill will take a while or for less obvious mistakes.
